app/src/main/java/com/infomaniak/drive/ui/addFiles/AddFileBottomSheetDialog.kt

```diff
@@ -110,12 +109,19 @@ class AddFileBottomSheetDialog : BottomSheetDialogFragment() {
         openCameraPermissions = CameraPermissions().apply {
             registerPermissions(this@AddFileBottomSheetDialog) { authorized -> if (authorized) openCamera() }
         }
-        uploadFilesPermissions = DrivePermissions().apply {
-            registerPermissions(this@AddFileBottomSheetDialog) { authorized -> if (authorized) uploadFiles() }
-        }
+
+        uploadFileHelper = UploadFileHelper(
+            fragment = this@AddFileBottomSheetDialog,
+            parentFolder = currentFolderFile,
+            onOpeningPicker = {
+                trackNewElement("uploadFile")
+                binding.documentUpload.isEnabled = false
+            },
+            onResult = { findNavController().popBackStack() },
+        )
```

Comment:
Is it possible to initialize uploadFileHelper using a `by lazy` ?

Reply:
Only if I dissociate the `parentFolder` argument of the constructor in an init function, but then we would need to avoid forgetting calling this init, thus I prefered to keep it in constructor